src/hotspot/cpu/aarch64/c1_Runtime1_aarch64.cpp

```diff
@@ -801,7 +801,7 @@ OopMapSet* Runtime1::generate_code_for(StubID id, StubAssembler* sasm) {
           __ andr(t1, t1, Klass::_lh_header_size_mask);
           __ sub(arr_size, arr_size, t1);  // body length
           __ add(t1, t1, obj);       // body start
-          __ initialize_body(t1, arr_size, 0, t2);
+          __ initialize_body(t1, arr_size, 0, t2, noreg);
```

Comment:
Is this the only place where we can end up with the the wrong temporaries in `initialize_body`? If so I think it would be better to add an assert in `initialize_body` and move the register saving here or else change the allocation of `t1`, `t2`, and `arr_size` above (see the earlier comment).